src/SAWScript/Prover/Exporter.hs

```diff
@@ -474,7 +474,7 @@ writeCoqCryptolModule inputFile outputFile notations skips = io $ do
                      , "From CryptolToCoq Require Import SAWCorePrelude."
                      , "Import SAWCorePrelude."
                      , "From CryptolToCoq Require Import CryptolPrimitivesForSAWCore."
-                     , "Import CryptolPrimitives."
+                     , "Import CryptolPrimitivesForSAWCore."
```

Comment:
I actually wonder if these lines are needed here... the generated code seems to have some repetition.

Reply:
If you mean that it seems redundant with the previous line, I don't think it is, this is essentially opening the namespace.

It is not a great design, but we had not spent time on keeping track of what identifiers were in scope or to what level they needed be qualified in a given scope.

Separately, I think a nicer design would be to have a set of wanted imports in the translation configuration, so as to guarantee the avoidance of duplicates. This is assuming that we would never care about the order of those imports.

Then the `postPreamble` could just contain weird ad-hoc stuff, if any.

Reply:
My generated code has sections that look like this:

```
(** Post-preamble section specified by you *)
From CryptolToCoq Require Import CryptolPrimitivesForSAWCore.
Import CryptolPrimitivesForSAWCore.
From CryptolToCoq Require Import SAWCorePrelude.
Import SAWCorePrelude.


(** Code generated by saw-core-coq *)

From CryptolToCoq Require Import SAWCorePrelude.
Import SAWCorePrelude.
From CryptolToCoq Require Import CryptolPrimitivesForSAWCore.
Import CryptolPrimitivesForSAWCore.
From CryptolToCoq Require Import CryptolPrimitivesForSAWCoreExtra.
```

There's some overlap between the two sections, and I'm not sure exactly why.

Reply:
As to the order of imports, they can matter in Coq, but we can probably arrange it so that we don't do any state changes in our modules that would make the order matter.

Reply:
Yeah, it's my bad, I had introduced:

https://github.com/GaloisInc/saw-script/blob/afbbf1bf89bf1be19e528d6b59c08399af7f5e0d/src/SAWScript/Prover/Exporter.hs#L466-L467

which I believe should have eliminated the following lines:

https://github.com/GaloisInc/saw-script/blob/afbbf1bf89bf1be19e528d6b59c08399af7f5e0d/src/SAWScript/Prover/Exporter.hs#L474-L477

Then we could do something similar for `CryptolPrimitivesForSAWCoreExtra`.